ImageFeed/Auth/WebViewViewController.swift

```diff
@@ -0,0 +1,105 @@
+import Foundation
+import UIKit
+import WebKit
+
+protocol WebViewViewControllerDelegate: AnyObject {
+    func webViewViewController(_ vc: WebViewViewController, didAuthenticateWithCode code: String)
+    func webViewViewControllerDidCancel(_ vc: WebViewViewController)
+}
+
+final class WebViewViewController: UIViewController {
+    
+    weak var delegate: WebViewViewControllerDelegate?
+    
+    @IBOutlet private weak var webView: WKWebView!
+    
+    @IBOutlet private weak var progressView: UIProgressView!
+    
+    @IBAction private func didTapBackButton(_ sender: Any?) {
+        delegate?.webViewViewControllerDidCancel(self)
+    }
+    
+    let UnsplashAuthorizeURLString = "https://unsplash.com/oauth/authorize"
+
+    override func viewWillAppear(_ animated: Bool) {
+        webView.addObserver(
+            self,
+            forKeyPath: #keyPath(WKWebView.estimatedProgress),
+            options: .new,
+            context: nil)
+        
+        updateProgress()
+    }
+    
+    override func viewDidLoad() {
+        var urlComponents = URLComponents(string: UnsplashAuthorizeURLString)!
+        urlComponents.queryItems = [
+        URLQueryItem(name: "client_id", value: AccessKey),
+        URLQueryItem(name: "redirect_uri", value: RedirectURI),
+        URLQueryItem(name: "response_type", value: "code"),
+        URLQueryItem(name: "scope", value: AccessScope)
+        ]
+        let url = urlComponents.url!
+        
+        let request = URLRequest(url: url)
+        webView.load(request)
+        
+        webView.navigationDelegate = self
+        
+        updateProgress()
+    }
+    
+    override func viewDidDisappear(_ animated: Bool) {
+        webView.removeObserver(
+            self,
+            forKeyPath: #keyPath(WKWebView.estimatedProgress),
+            context: nil)
+    }
+    
+    private func code(from navigationAction: WKNavigationAction) -> String? {
+        if
+            let url = navigationAction.request.url,
+            let urlComponents = URLComponents(string: url.absoluteString),
+            urlComponents.path == "/oauth/authorize/native",
+            let items = urlComponents.queryItems,
+            let codeItems = items.first(where: { $0.name == "code" })
+        {
+            return codeItems.value
+        } else {
+            return nil
+        }
```

Comment:
этот код можно переписать на guard. Но и твой вариант правильный
```suggestion
    private func code(from navigationAction: WKNavigationAction) -> String? {
        guard
            let url = navigationAction.request.url,
            let urlComponents = URLComponents(string: url.absoluteString),
            urlComponents.path == "/oauth/authorize/native",
            let items = urlComponents.queryItems,
            let codeItems = items.first(where: { $0.name == "code" })
        else {
            return nil
        } 
        
        return codeItems.value
```